bindings/java/src/main/com/apple/foundationdb/TenantManagement.java

```diff
@@ -210,5 +211,109 @@ public static CompletableFuture<Void> deleteTenant(Database db, Tuple tenantName
 		return deleteTenant(db, tenantName.pack());
 	}
 
+
+	/**
+	 * Lists all tenants in between the range specified. The number of tenants listed can be restricted.
+	 *
+	 * @param db The database used to create a transaction for listing the tenants.
+	 * @param begin The beginning of the range of tenants to list.
+	 * @param end The end of the range of the tenants to list.
+	 * @param limit The maximum number of tenants to return from this request.
+	 * @return an iterator where each item is a KeyValue object where the key is the tenant name
+	 * and the value is the unprocessed JSON string containing the tenant's metadata
+	 */
+	public static CloseableAsyncIterator<KeyValue> listTenants(Database db, byte[] begin, byte[] end, int limit) {
+		return listTenants_internal(db.createTransaction(), begin, end, limit);
+	}
+
+	/**
+	 * Lists all tenants in between the range specified. The number of tenants listed can be restricted.
+	 * This is a convenience method that generates the begin and end ranges by packing two {@code Tuple}s.
+	 *
+	 * @param db The database used to create a transaction for listing the tenants.
+	 * @param begin The beginning of the range of tenants to list.
+	 * @param end The end of the range of the tenants to list.
+	 * @param limit The maximum number of tenants to return from this request.
+	 * @return an iterator where each item is a KeyValue object where the key is the tenant name
+	 * and the value is the unprocessed JSON string containing the tenant's metadata
+	 */
+	public static CloseableAsyncIterator<KeyValue> listTenants(Database db, Tuple begin, Tuple end, int limit) {
+		return listTenants_internal(db.createTransaction(), begin.pack(), end.pack(), limit);
```

Comment:
It's been a while since we had this discussion; where did we land on the idea of having the call with a `Database` object be multi-transactional?

Reply:
Discussed to create a larger, generic wrapper for multi-transactional handling in a separate PR